src/OrleansEventSourcing/EventStorage/LogViewAdaptor.cs

```diff
@@ -0,0 +1,360 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orleans.EventSourcing.Common;
+using Orleans.LogConsistency;
+using Orleans.Runtime;
+using Orleans.Storage;
+
+namespace Orleans.EventSourcing.EventStorage
+{
+
+    /// <summary>
+    /// A log view adaptor that is based on an event store
+    /// </summary>
+    /// <typeparam name="TState">The type of the log view, or grain state.</typeparam>
+    /// <typeparam name="TEvent">The type of the events (usually just object)</typeparam>
+    class EventStoreLogViewAdaptor<TState, TEvent> : PrimaryBasedLogViewAdaptor<TState, TEvent, SubmissionEntryWithGuid<TEvent>>
+        where TState : class, new() where TEvent : class
+    {
+
+        public EventStoreLogViewAdaptor(
+            ILogViewAdaptorHost<TState, TEvent> host, 
+            ILogConsistencyProvider provider,
+            TState initialstate, 
+            ILogConsistencyProtocolServices services)  : base(host, initialstate, services)
+        {
+            this.provider = provider as LogConsistencyProvider;
+
+            if (this.provider == null)
+                throw new ArgumentException("provider must be an event store", nameof(provider));
+
+        }
+
+
+        private LogConsistencyProvider provider;
+
+        private TState ConfirmedStateInternal;
+        private int ConfirmedVersionInternal;
```

Comment:
ok. suffix "Internal" was just so the name does not clash with the public property. But that is no longer an issue when using camel case anyway.